VMEncryption/main/DiskUtil.py

```diff
@@ -647,38 +647,40 @@ def get_encryption_status(self):
         }
 
         mount_items = self.get_mount_items()
+        device_items = self.get_device_items(None)
+        device_items_dict = {di.mount_point: di for di in device_items}
 
         os_drive_encrypted = False
         data_drives_found = False
         all_data_drives_encrypted = True
+
+        if self.is_os_disk_lvm():
+            grep_result = self.command_executor.ExecuteInBash('pvdisplay | grep {0}'.format(self.get_osmapper_path()),
+                                                              suppress_logging=True)
+            if grep_result == 0 and not os.path.exists('/volumes.lvm'):
+                self.logger.log("OS PV is encrypted")
+                os_drive_encrypted = True
+
+        special_azure_devices_to_skip = self.get_azure_devices()
         for mount_item in mount_items:
-            if mount_item["fs"] in ["ext2", "ext4", "ext3", "xfs"] and \
-                not "/mnt" == mount_item["dest"] and \
-                not "/" == mount_item["dest"] and \
-                not "/oldroot/mnt/resource" == mount_item["dest"] and \
-                not "/oldroot/boot" == mount_item["dest"] and \
-                not "/oldroot" == mount_item["dest"] and \
-                not "/mnt/resource" == mount_item["dest"] and \
-                not "/boot" == mount_item["dest"]:
+            device_item = device_items_dict.get(mount_item["dest"])
 
+            if device_item is not None and \
+               mount_item["fs"] in ["ext2", "ext4", "ext3", "xfs"] and \
+               self.is_data_disk(device_item, special_azure_devices_to_skip):
                 data_drives_found = True
```

Comment:
indentation seems wrong? 
do we have unit tests added for this?

Reply:
We don't yet have unit tests for this file. Some more refactoring is needed before we can set those up.